database/meta.go

```diff
@@ -95,27 +95,31 @@ func (m *MetaDB) SetRollingSettings(db string, chunk int, numchunks int) error {
 	ssn := m.dbHandle.Copy()
 	defer ssn.Close()
 
+	selector := bson.M{"name": db}
+	update := bson.M{
+		"rolling": true,
+		"current_chunk": chunk,
+		"total_chunks": numchunks,
+	}
+
 	if !result.Rolling {
 		m.log.Infof("The dataset [ %s ] is being converted to a rolling dataset.", db)
-	} else if result.TotalChunks != numchunks {
-		m.log.Warnf("The total chunk size for existing rolling dataset [ %s ] was set to [ %d ] and is being changed to [ %d ].",
+	} else if result.TotalChunks < numchunks {
+		m.log.Warnf("The total chunk size for existing rolling dataset [ %s ] was set to [ %d ] and is being increased to [ %d ].",
+			db, result.TotalChunks, numchunks)
+
+		// if the total chunks increaase grow cid_list while preserving original entries
+		for i := result.TotalChunks; i < numchunks; i++ {
```

Comment:
This is based on the code from `SetChunk` later in this file. I originally tried to just call `SetChunk` in a loop but these both lock the mutex so it was a deadlock. Plus it would have generated a bunch of mongo queries instead of only one.

Technically, I don't think we need to do this. I was using it before without this and it worked fine. In general the `cid_list` gets incremented naturally for every new chunk that is added after the number of chunks are increased. You could feasibly import a chunk that is beyond what has been imported before (e.g. there are 12 existing chunks and you import into chunk 23). Without this for loop what happens is you get this:
```json
"cid_list" : [ { "set" : true }, { "set" : true }, { "set" : false }, { "set" : false }, { "set" : false }, { "set" : false }, { "set" : false }, { "set" : false }, { "set" : false }, { "set" : false }, { "set" : false }, { "set" : false }, null, null, null, null, null, null, null, null, null, null, null, { "set" : true } ]
```
Note that Mongo just set the 24th element to the right value and filled in every array value between with `null`s. With the current code this shouldn't break because we're just checking if a chunk `set` = `true`. But I thought it was more explicit and more predictable to have the values be initialized to `{set: false}` here. I'm just trying to do this in the way least likely to break anything now or in the future.